py/mpstate.h

```diff
@@ -87,6 +87,7 @@ typedef struct _mp_state_mem_area_t {
     byte *gc_pool_end;
 
     size_t gc_last_free_atb_index;
+    size_t gc_high_watermark; // The block ID of the highest block allocated in the area
```

Comment:
I suggest renaming this variable to `gc_last_used_block` or `gc_last_allocated_block` or `gc_last_non_free_block` (because watermark could mean a mark for other statistics/counters).

Reply:
+1 to `gc_last_used_block`.